### PR TITLE
feat: rename sample_degseq() method and add the edge.switching.simple method

### LINKED ISSUE #876: sample_degseq() method parameter values should be renamed to match C core
The allowed value for `sample_degseq`'s `method` parameter should be renamed to match the 0.10 C core. Old names can be kept for compatibility. The renaming can be done now, even before the 0.10 upgrade.

See the C docs:

https://igraph.org/c/html/latest/igraph-Generators.html#igraph_degree_sequence_game

The R docs should be updated. Feel free to copy and paste from the C docs.

Renaming should be done like so:

 - `simple` -> `configuration`
 - `simple.no.multiple` -> `fast.heur.simple`
 - `simple.no.multiple.uniform` -> `configuration.simple`
 - `vl` does not need renaming
 - when upgrading to 0.10, there will be a new method called `edge.switching.simple`.

The new names are rationalized. When they contain the word "simple", it means that they produce simple graphs (not multigraphs).

If someone will do the renaming, and deprecates the old names, I can do the doc update.

### PR DESCRIPTION
**Pull request by maelle**
Fix  #876

See @szhorvat, a new feature exposed after all the boring refactoring PRs. :grin: :innocent: Some questions

- where do the numbers corresponding to the method come from? I assumed the new one was 4.
- what could be a better test for the new method?

### CODE REVIEW

File: R/games.R [hunk 1]
```diff
@@ -909,13 +912,30 @@ random.graph.game <- erdos.renyi.game
...
+
   method1 <- switch(method,
-    "simple" = 0,
+    "configuration" = 0,
     "vl" = 1,
-    "simple.no.multiple" = 2,
-    "simple.no.multiple.uniform" = 3
+    "fast.heur.simple" = 2,
+    "configuration.simple" = 3,
+    "edge.switching.simple" = 4 #TODO: check number and see where it comes from
```
Comment (maelle) [inline R/games.R:938]: ideally we should make a note of where the numbers come from, for the next time someone needs to touch this part of the file.

**Comment by szhorvat** [comment]
> where do the numbers corresponding to the method come from? I assumed the new one was 4.

For enum values in general, you need to look in `include/igraph_constants.h` in the C core:

https://github.com/igraph/igraph/blob/master/include/igraph_constants.h#L94

Enums in C are "named values" represented internally as integers. The numbers are integers. Notice that the first value is explicitly assigned `0`. Anything after that that does not have an explicit assignment increments by 1. So yes, the new values is `4`.

In the majority of cases we start at `0`, but there are a handful of exceptions, so it's good to have a look at the sources.

 > what could be a better test for the new method?

Checking that the degree sequence of the result is what we input is the best we can do here. But your test has a problem: you generate an undirected graph then you use both in- and out-degrees, which are for directed graphs. Add `directed=T` in `sample_gnm`.

It might be useful, but not essential, to also test the undirected case, with an undirected starting point and undirected degrees.

**Comment by maelle** [comment]
> It might be useful, but not essential, to also test the undirected case, with an undirected starting point and undirected degrees.

Do you have an example at hand?

**Comment by maelle** [comment]
I made edits @szhorvat. Thanks for your feedback!

**Comment by szhorvat** [comment]
> Do you have an example at hand?

 - Generate a `directed=FALSE` graph
 - Take `degrees(graph, mode="all")`
 - Use this as a single input to `sample_degseq()`
 - Check that the result has the same degrees, again with `degrees(graph, mode="all")`

**Comment by szhorvat** [comment]
Could you please update the docs by lifting the descriptions of the five methods from the C docs here? https://igraph.org/c/html/latest/igraph-Generators.html#igraph_degree_sequence_game  The current R docs have misleading statements.  It's quickest to copy-paste the method description from C, and fix up the math so it actually renders are proper math.

**Comment by szhorvat** [comment]
I improved the C docs a bit in this commit: 563d594a0bc5aca462a2df64a968fa0cb2bd9bcd

**Comment by szhorvat** [comment]
I made some more (independent) doc updates, so I'll just leave this link here instead of linking to comments:

https://github.com/igraph/igraph/blob/master/src/games/degree_sequence.c#L671

These updates did not affect the text I suggested you copy, but there's additional information there that might be worth copying (such as the "see also" section).

**Comment by maelle** [comment]
Is this PR good as is? I should put https://github.com/igraph/rigraph/issues/1317 nearer the top of my list :wink:

**Comment by szhorvat** [comment]
Yes, let's merge this. Further doc improvements are always possible, and should not block this PR.

**Pull request merged**